packages/http-client/src/index.ts

```diff
@@ -701,6 +714,44 @@ export class HttpClient {
     return agent
   }
 
+  private _getProxyAgentDispatcher(parsedUrl: URL, proxyUrl: URL): ProxyAgent {
+    let proxyAgent
+
+    if (this._keepAlive) {
+      proxyAgent = this._proxyAgentDispatcher
+    }
+
+    // if agent is already assigned use that agent.
+    if (proxyAgent) {
+      return proxyAgent
+    }
+
+    const usingSsl = parsedUrl.protocol === 'https:'
+
+    // This is `useProxy` again, but we need to check `proxyURl` directly for TypeScripts's flow analysis.
+    if (proxyUrl && proxyUrl.hostname) {
```

Comment:
Any reason not to just use optional chaining here?

```suggestion
    if (proxyUrl?.hostname) {
```

Reply:
Ah, actually `proxyUrl` is defined as a **non-nullable** `URL`, so you don't even need to check it, it's guaranteed by the type system to be there. 

```suggestion
    if (proxyUrl.hostname) {
```

Reply:
Resolved